src/workloads/scale/UpdateMillionDocsInTxn.yml

```diff
@@ -0,0 +1,78 @@
+# This workload is developed to test the amount of time it takes to update a million documents in
+# a single replica set transaction. At the moment, the average time taken raises above the default
+# limit, so until we add a way to manually increase a transaction's lifetime, we must raise the
+# lifetime of all transactions.
+
+SchemaVersion: 2018-07-01
+Owner: "@mongodb/sharding"
+
+Actors:
+- Name: Loader
+  Type: Loader
+  Threads: 1
+  Phases:
+  - ExternalPhaseConfig:
+      Path: &PhasePath ../../phases/scale/DesignDocWorkloadPhases.yml
+      Key: Loader
+      Parameters:
+        CollectionCount: 1
+        DocumentCount: 1e6
+        BatchSize: 100000
+    Repeat: 1
+    Threads: 1
+  - &Nop {Nop: true}
+  - *Nop
+
+- Name: RaiseTransactionLifetimeLimit
+  Type: AdminCommand
+  Threads: 1
+  Phases:
+  - Repeat: 1
+    Operation:
+      OperationName: RunCommand
+      OperationCommand:
+        setParameter: 1
+        transactionLifetimeLimitSeconds: 120
+  - *Nop
+  - Repeat: 1
+    Operation:
+      OperationName: RunCommand
+      OperationCommand:
+        setParameter: 1
+        transactionLifetimeLimitSeconds: 60
+
+- Name: UpdateDocumentsInTransaction
+  Type: CrudActor
+  Database: test
+  Phases:
+  - *Nop
+  - MetricsName: TransactionUpdate
+    Repeat: 1
```

Comment:
Are you sure you only want to repeat this 1 time? I'd be worried about how noisy this may be. More typically we'll see at least 100-1k repeats. It's okay if each iteration takes a while we want good data. Your call here but @dalyd may have more input.

Reply:
How long does the transaction take, and how repeatable are the results?

Reply:
Each transaction takes about 90 seconds to run... not sure how many times would be acceptable for this test.

Reply:
Let's do a few of them (that's purposefully vague). Then we can get a distribution also. We want to keep tasks under 1 hour. We should be well below that now. If the test runs for a 5-10 minutes that seems good to me. Note that there are some processes at a 60 second granularity in the server, so doing a few minutes helps average that out.

Reply:
Setting it to repeat five times.